### PR TITLE
[RF] Multiple small fixes to `RooHist`

### PR DESCRIPTION
**Pull request by guitargeek**
More detail in the commit descriptions.

Closes JIRA ticket ROOT-5805:
https://sft.its.cern.ch/jira/browse/ROOT-5805

### CODE REVIEW

File: roofit/roofitcore/src/RooHist.cxx [hunk 1]
```diff
@@ -621,7 +592,7 @@ bool RooHist::hasIdenticalBinning(const RooHist& other) const
     GetPoint(i,x1,y1) ;
     other.GetPoint(i,x2,y2) ;
 
-    if (fabs(x1-x2)>1e-10) {
+    if (std::abs(x1-x2)>1e-10) {
```
Comment (lmoneta) [inline roofit/roofitcore/src/RooHist.cxx:595]: Is the condition correct in case of x1 and x2 are very large ?

**Review by lmoneta — APPROVED** [review]
LGTM, thanks you for the fixes !
Maybe some of the conditions with  values hard-coded could be improved but it is not urgent.

File: roofit/roofitcore/src/RooHist.cxx [hunk 2]
```diff
@@ -394,7 +382,7 @@ Int_t RooHist::roundBin(double y)
     return 0;
   }
   Int_t n= (Int_t)(y+0.5);
-  if(fabs(y-n)>1e-6) {
+  if(std::abs(y-n)>1e-6) {
```
[the remark this answers is not in the document]
Comment (guitargeek) [reply roofit/roofitcore/src/RooHist.cxx:385]: Yes, if you look up a few lines you can see the function returns before when y is negative

File: roofit/roofitcore/src/RooHist.cxx (hunk 1 above)
Comment (guitargeek) [reply roofit/roofitcore/src/RooHist.cxx:595]: Thank for spotting this, good point!

Indeed, as the goal of this function is to figure out if two `RooHists` have the same bin centers within some small epsilon, this epsilon should better depend on the overall scale of the x values. I think this is well represented by the average bin width, so I'll change this to:

```c++
if (std::abs(x1-x2) > 1e-10 * _nominalBinWidth) {
```

**Comment by guitargeek** [comment]
Thanks for the review, especially for looking also at the trivial changes and giving suggestions to improve the code!

**Pull request merged**